crates/pico_core/src/node.rs

```diff
@@ -20,24 +20,26 @@ impl DerivedNodeId {
 }
 
 pub struct DerivedNode<Db: Database + ?Sized> {
-    pub time_verified: Epoch,
-    pub time_updated: Epoch,
     pub dependencies: Vec<Dependency>,
```

Comment:
Just idly thinking... I feel like a *lot* of functions will have one (or max a few) dependencies. So, maybe something like https://docs.rs/smallvec/latest/smallvec/ will help save a memory lookup in the common case.

In particular, I imagine we have many functions like `ast_without_locations(db, ast_id: SomeId) { db.lookup(ast_id).remove_locations() }`. If we have something like `hover_info`, which depends on `full_ast`, every (even irrelevant) change will cause `hover_info` to be recalculated. It's better to have that intermediate function.

Let's do that as a follow up, though